HPXMLtoOpenStudio/tests/test_validation.rb

```diff
@@ -0,0 +1,264 @@
+            this_parents = XMLHelper.get_elements(parent_element, child_element_without_predicates)
+          end
+
+          this_parents.each do |e|
+            XMLHelper.add_element(e, this_child_name, this_child_value)
+          end
+        end
+      end
+
+      # Ruby validator validation
+      _test_ruby_validation(hpxml_doc, expected_error_message)
+      # Schematron validation
+      _test_schematron_validation(@stron_doc, hpxml_doc.to_xml, expected_error_message)
+    end
+
+    puts
+  end
+
+  private
+
+  def _test_schematron_validation(stron_doc, hpxml, expected_error_msgs = nil)
+    # load the xml document you wish to validate
+    xml_doc = Nokogiri::XML hpxml
+    # validate it
+    results = stron_doc.validate xml_doc
+    # assertions
+    if expected_error_msgs.nil?
+      assert_empty(results)
+    else
+      idx_of_interest = results.index { |i| i[:message].gsub(': ', [': ', i[:context_path].gsub('h:', '').concat(': ').gsub('/*: ', '')].join('')) == expected_error_msgs }
+      error_msg_of_interest = results[idx_of_interest][:message].gsub(': ', [': ', results[idx_of_interest][:context_path].gsub('h:', '').concat(': ').gsub('/*: ', '')].join(''))
+      assert_equal(expected_error_msgs, error_msg_of_interest)
+    end
+  end
+
+  def _test_ruby_validation(hpxml_doc, expected_error_msgs = nil)
+    # Validate input HPXML against EnergyPlus Use Case
+    results = EnergyPlusValidator.run_validator(hpxml_doc)
+    if expected_error_msgs.nil?
+      assert_empty(results)
+    else
+      idx_of_interest = results.index { |i| i == expected_error_msgs }
+      error_msg_of_interest = results[idx_of_interest]
+      assert_equal(expected_error_msgs, error_msg_of_interest)
+    end
+  end
+
+  def _test_schema_validation(hpxml_doc, xml)
+    # TODO: Remove this when schema validation is included with CLI calls
+    schemas_dir = File.absolute_path(File.join(@root_path, 'HPXMLtoOpenStudio', 'resources'))
+    errors = XMLHelper.validate(hpxml_doc.to_xml, File.join(schemas_dir, 'HPXML.xsd'), nil)
+    if errors.size > 0
+      puts "#{xml}: #{errors}"
+    end
+    assert_equal(0, errors.size)
+  end
+
+  def _get_hpxml_doc(target_xpath, mode)
+    @hpxml_docs.values.each do |hpxml_doc|
+      parent_element = target_xpath[0] == '' ? hpxml_doc : XMLHelper.get_element(hpxml_doc, target_xpath[0])
+      next if parent_element.nil?
+
+      child_elements = target_xpath[1]
+
+      if mode == 'deletion'
+        end_index = -1
+      elsif mode == 'addition'
+        end_index = -2
+      end
+
+      found_children = true
+      child_elements[0..end_index].each do |child_element|
+        next if XMLHelper.has_element(parent_element, child_element)
+
+        found_children = false
+        break
+      end
+      next unless found_children
+
+      return _deep_copy_object(hpxml_doc)
+    end
+
+    fail "Could not find HPXML file for target_xpath: #{target_xpath}."
+  end
+
+  def _get_element_name(assertion)
```

Comment:
The method is called "_get_element_name" but it returns an array. Please update the method name and provide a brief comment describing what the method does.